server/modules/selva/module/traversal.h

```diff
@@ -0,0 +1,500 @@
+                        data_len = raw_value_len;
+                        type = ORDERED_ITEM_TYPE_TEXT;
+                        break;
+                    }
+                }
+            }
+        }
+    } else if (obj_type == SELVA_OBJECT_DOUBLE) {
+        err = SelvaObject_GetDouble(obj, order_field, &d);
+        if (!err) {
+            type = ORDERED_ITEM_TYPE_DOUBLE;
+        }
+    } else if (obj_type == SELVA_OBJECT_LONGLONG) {
+        long long v;
+
+        err = SelvaObject_GetLongLong(obj, order_field, &v);
+        if (!err) {
+            d = (double)v;
+            type = ORDERED_ITEM_TYPE_DOUBLE;
+        }
+    }
+
+    size_t final_data_len = data_len;
+    locale_t locale = 0;
+
+    if (type == ORDERED_ITEM_TYPE_TEXT && data_len > 0) {
+        locale = SelvaLang_GetLocale(data_lang, strlen(data_lang));
+        final_data_len = strxfrm_l(NULL, data, 0, locale);
+    }
+
+    item = RedisModule_PoolAlloc(ctx, sizeof(struct FindCommand_OrderedItem) + final_data_len + 1);
+    if (!item) {
+        /*
+         * Returning NULL in case of ENOMEM here should be fairly ok as we can
+         * assume that Redis will free everything we allocated and opened before
+         * this point. Although it's possible that there is not enough memory to
+         * do the cleanup but there is nothing we could do better neither here.
+         */
+        return NULL;
+    }
+
+    memcpy(item->id, EMPTY_NODE_ID, SELVA_NODE_ID_SIZE);
+    item->type = type;
+    if (type == ORDERED_ITEM_TYPE_TEXT && data_len > 0) {
+        strxfrm_l(item->data, data, final_data_len + 1, locale);
+    }
+    item->data_len = data_len;
+    item->d = d;
+    item->data_obj = obj;
+
+cleanup:
+    return item;
+}
+
+static int fields_contains(struct SelvaObject *fields, const char *field_name_str, size_t field_name_len) {
+    void *iterator;
+    const SVector *vec;
+
+    iterator = SelvaObject_ForeachBegin(fields);
+    while ((vec = (SVector *)SelvaObject_ForeachValue(fields, &iterator, NULL, SELVA_OBJECT_ARRAY))) {
+        struct SVectorIterator it;
+        const RedisModuleString *s;
+
+        SVector_ForeachBegin(&it, vec);
+        while ((s = SVector_Foreach(&it))) {
+            TO_STR(s);
+
+            if (s_len == field_name_len && !strcmp(s_str, field_name_str)) {
+                return 1;
+            }
+        }
+    }
+
+    return 0;
+}
+
+static int get_skip(enum SelvaModify_HierarchyTraversal dir) {
+    switch (dir) {
+     /*
+      * Find needs to skip the head node of the traverse for some types as we
+      * are not interested in the node we already know.
+      */
+    case SELVA_HIERARCHY_TRAVERSAL_BFS_ANCESTORS:
+    case SELVA_HIERARCHY_TRAVERSAL_BFS_DESCENDANTS:
+    case SELVA_HIERARCHY_TRAVERSAL_DFS_ANCESTORS:
+    case SELVA_HIERARCHY_TRAVERSAL_DFS_DESCENDANTS:
+    case SELVA_HIERARCHY_TRAVERSAL_DFS_FULL:
+        return 1;
+    default:
+        return 0;
+    }
+}
+
```

Comment:
```suggestion
int SelvaTraversal_ParseOrder(
        const RedisModuleString **order_by_field,
        enum hierarchy_result_order *order,
        const RedisModuleString *txt,
        const RedisModuleString *fld,
        const RedisModuleString *ord);
        
```

+ others GH is soo broken with big diffs.